cloudinit/distros/opensuse.py

```diff
@@ -281,3 +285,143 @@ def preferred_ntp_clients(self):
                 ]
 
         return self._preferred_ntp_clients
+
+    def _write_network_state(
+        self,
+        network_state,
+        renderer: Renderer,
+        netconfig: Optional[dict] = None,
+    ):
+        super()._write_network_state(network_state, renderer)
+        netconfig = netconfig or {}
+        netconfig_ver = netconfig.get("version")
+        if netconfig_ver == 1:
+            self._write_routes_v1(netconfig)
+        elif netconfig_ver == 2:
+            self._write_routes_v2(netconfig)
+        else:
+            LOG.warning(
+                "unsupported or missing netconfig version, not writing routes"
+            )
+
+    def _write_routes_v1(self, netconfig: dict):
+        """Write route files, not part of the standard distro interface"""
+        # Due to the implementation of the sysconfig renderer default routes
+        # are setup in ifcfg-* files. But this does not work on SLES or
+        # openSUSE https://bugs.launchpad.net/cloud-init/+bug/1812117
+        # this is a very hacky way to get around the problem until a real
+        # solution is found in the sysconfig renderer
+        device_configs = netconfig.get("config", [])
+        default_nets = ("::", "0.0.0.0")
+        for config in device_configs:
+            if_name = config.get("name")
+            subnets = config.get("subnets", [])
+            config_routes = ""
+            has_default_route = False
+            seen_default_gateway = None
+            for subnet in subnets:
+                # Render the default gateway if it is present
+                gateway = subnet.get("gateway")
+                if gateway:
+                    config_routes += " ".join(["default", gateway, "-", "-\n"])
+                    has_default_route = True
+                    if not seen_default_gateway:
+                        seen_default_gateway = gateway
+                # Render subnet routes
+                routes = subnet.get("routes", [])
+                for route in routes:
+                    dest = route.get("destination") or route.get("network")
+                    if not dest or dest in default_nets:
+                        dest = "default"
+                        if not has_default_route:
+                            has_default_route = True
+                    if dest != "default":
+                        netmask = route.get("netmask")
+                        if netmask:
+                            prefix = ipv4_mask_to_net_prefix(netmask)
```

Comment:
...and then changes the code here to:
```
                            if net.is_ipv4_network(netmask):
                                prefix = net.ipv4_mask_to_net_prefix(netmask)
                            if net.is_ipv6_network(netmask):
                                prefix = net.ipv6_mask_to_net_prefix(netmask)
```
This is currently being tested by the customer that found this issue. This happened because the customer config includes a `netmask` setting for an IPv6 network, which was previously not considered.

Reply:
Hey @rjschwei, did the customer find any issue with this implementation? If not then I'll update this PR to reflect your suggestions.

Reply:
@holmanb the updated patch is not yet released but has been verified to work.